### ISSUE TITLE
Slow startup time on Apple Silicon using npm.

### ISSUE THREAD

**Issue by tplk** [evilmartians/lefthook#287]
Hello! I'm experiencing very slow startup times using `lefthook` with `npm`. The summary logs sub 10ms execution times for the scripts themselves, but in total it takes up to 1 second to execute.

I've been able to track the issue down to the lefthook's git hook shell script which locates the binary. It seems like on the ARM platform there's no binary installed to `node_modules/@arkweid/lefthook/bin/lefthook` and the script uses `npx @arkweid/lefthook` as a fallback to run the program.

<img width="554" alt="image" src="https://user-images.githubusercontent.com/20531968/175679484-0bff45c0-cf5f-4b22-84b8-87186a525114.png">

There are a couple fixes which both worked great for me:
- symlinking `node_modules/@arkweid/lefthook/bin/index.js` to `node_modules/@arkweid/lefthook/bin/lefthook`,
- adding `node_modules/.bin/lefthook` path check into the hook's conditions (the result can be seen on the screenshot below).

<img width="792" alt="image" src="https://user-images.githubusercontent.com/20531968/175695769-e3c9a1d3-0642-4e92-b59d-ec95cd80414f.png">

As you can see I get `0.81s` by default and `0.06s` after the patch.

```patch
17a18,20
>   elif test -f "$dir/node_modules/.bin/lefthook"
>   then
>     eval "$dir/node_modules/.bin/lefthook $@"
```

Would it be a reasonable fix?

**Comment by mrexox** [comment]
Hey! Good catch, thank you for reporting!

Can you try `lefthook` npm package with [1.0.2 version](https://www.npmjs.com/package/lefthook)? I see you are using `@arkweid/lefthook` package that is not maintained anymore.

In 1.0.2 I've added [straight call](https://github.com/evilmartians/lefthook/blob/v1.0.2/internal/templates/hook.tmpl#L26-L28) to `lefthook` executable inside node_modules.

Anyway you fix seems reasonable and I will apply it :+1:

**Comment by mrexox** [comment]
Hm, I have a concern. We can't be sure that `node_modules/.bin/lefthook` was provided by `lefthook` package. I mean there is a risk for a malware. Please, try the latest `lefthook` npm package, it must work faster.

`lefthook install --force` will regenerate the git hooks and update the script.

**Comment by tplk** [comment]
>  I see you are using @arkweid/lefthook package that is not maintained anymore.

Hm, you're totally right. The last time I've used the package it was available under that name and I missed that it's an older version. I think I might've gotten it from one of the articles linked in the README this time.

I've switched to `@evilmartians/lefthook` and everything works great! To be exact not only it works but also locates the correct binary based on the system arch. So the direct call to `bin/index.js` is not even needed).

Thanks! I'll close the issue now.
